src/simpleapi.jl

```diff
@@ -62,58 +65,55 @@ function get(ctx::KuberContext, O::Symbol, name::String, apiversion::Union{Strin
     isempty(ctx.apis) && set_api_versions!(ctx; max_tries=max_tries)
 
     apictx = _get_apictx(ctx, O, apiversion)
-    try
-        apicall = eval(Symbol("read$O"))
+    if (apicall = _api_function("read$O")) !== nothing
         @repeat max_tries try
             return apicall(apictx, name; kwargs...)
         catch e
             @retry if isa(e, IOError)
-                @debug("Retrying ", "read$O")
+                @debug("Retrying ", nameof(apicall))
                 sleep(2)
             end
         end
-    catch ex
-        isa(ex, UndefVarError) || rethrow()
-        apicall = eval(Symbol("readNamespaced$O"))
+    elseif (apicall = _api_function("readNamespaced$O")) !== nothing
         @repeat max_tries try
             return apicall(apictx, name, ctx.namespace; kwargs...)
         catch e
             @retry if isa(e, IOError)
-                @debug("Retrying ", "readNamespaced$O")
+                @debug("Retrying ", nameof(apicall))
                 sleep(2)
             end
         end
+    else
+        throw(ArgumentError("No API functions could be located using :$O"))
     end
 end
 
 function get(ctx::KuberContext, O::Symbol, apiversion::Union{String,Nothing}=nothing; label_selector=nothing, namespace::Union{String,Nothing}=ctx.namespace, max_tries::Integer=1)
     isempty(ctx.apis) && set_api_versions!(ctx; max_tries=max_tries)
 
     apictx = _get_apictx(ctx, O, apiversion)
-    try
-        apiname = "list$O"
-        (namespace === nothing) && (apiname *= "ForAllNamespaces")
-        apicall = eval(Symbol(apiname))
+    apiname = "list$O"
+    namespace === nothing && (apiname *= "ForAllNamespaces")
+    if (apicall = _api_function(apiname)) !== nothing
         @repeat max_tries try
             return apicall(apictx; labelSelector=label_selector)
         catch e
             @retry if isa(e, IOError)
-                @debug("Retrying ", apiname)
+                @debug("Retrying ", nameof(apicall))
                 sleep(2)
             end
         end
-    catch ex
-        isa(ex, UndefVarError) || rethrow()
-        (namespace === nothing) && rethrow()
```

Comment:
Note that if `namespace = nothing` and the API function `list$(O)ForAllNamespaces` did not exist then this would cause `UndefVarError: list$(O)ForAllNamespaces` to be rethrown. The current implementation will attempt to call `listNamespaces$O` and a `MethodError` would be thrown which may be better.

I could do some additional iteration here if necessary